### PR TITLE
feat(perf-issues): Skip truncated queries in slow db span detection

### PR DESCRIPTION
**Pull request by 0Calories**
If we detect on truncated queries, we can't guarantee that the fingerprint is unique to the same query. This PR changes the slow span detector to skip DB spans with queries that are truncated

### CODE REVIEW

File: src/sentry/utils/performance_issues/performance_detection.py [hunk 1]
```diff
@@ -572,6 +570,21 @@ def visit_span(self, span: Span):
...
             )
 
+    @classmethod
+    def is_span_eligible(cls, span: Span) -> bool:
+        description = span.get("description", None)
+        if not description:
+            return False
+
+        description = description.strip()
+        if description[:6].upper() != "SELECT":
```
Comment (DominikB2014) [inline src/sentry/utils/performance_issues/performance_detection.py:580]: I think startswith would be the same?

**Review by DominikB2014 — APPROVED** [review]
lgtm!

File: src/sentry/utils/performance_issues/performance_detection.py (hunk 1 above)
Comment (0Calories) [reply src/sentry/utils/performance_issues/performance_detection.py:580]: Sometimes the query starts with a lowercase `select` so I can't really use `startswith` here (I guess I could but then I'd need to check with `upper()` and `lower()`)

Comment (DominikB2014) [reply src/sentry/utils/performance_issues/performance_detection.py:580]: Fair enough! I personally like doing upper on the whole description, will leave it to you tho!

**Pull request merged**